server/models/handlers/appointment.js

```diff
@@ -24,7 +24,18 @@ async function getAppointmentDetails(studentId, workerId, status) {
     return appointmentModel.getAppointmentDetails(studentId, workerId, status);
 }
 
+// Attempts to cancel all the worker appointments and updates their availability to unavailable for the entire day
+// or for a specified period.
+// Returns true if the worker appointments and their times were successfully updated and false if error encountered.
+async function cancelWorkerAppointments(workerId, startTime, endTime) {
```

Comment:
What do you think of the parameters for this functionality? Is it necessary to split out startTime and startDate? Or is there a way to control the inputs to always be of a certain format and then apply a split once the data is passed in?

Reply:
My reasoning for keeping the date and time together is to simplify the amount of parameters and increase the flexibility of this endpoint. To elaborate, this format currently allows for a variety of startTime and endTimes to be specified and is consistent with how the SQL query would be made. If these were split into the separate date and time, then these parameters would double from 2 to 4: startDate, endDate, startTime, endTime and may require additional logic in the SQL query. 

The query parameters are currently controlled through the Joi validation as date validation of these values occur during the initialization of the private endpoint. Thus, this would ensure that the dates passed in are of a certain format (ISO dates). Currently, I plan to leverage the moment library to apply the split (once the data is passed in).

Reply:
I appreciate the thought put into this! As long as there's a clear path forward I'm on board. I hear your points about doubling the number of parameters, I agree.

Reply:
Hi Peter,
Upon further look, I think we could leverage the split to reduce the complexity of our SQL queries. Although it would be better to simplify the number of parameters, we could potentially leverage the numbering of workerTimeslotIds for filtering use. Since the front end would be displaying these workerTimeslots as a dropdown, it would be easier to pass these ids to the backend when specifying the start and end dates, which would decrease the complexity of the SQL queries. I shall make this change.